R/get_source_expressions.R

```diff
@@ -172,23 +185,27 @@ lint_parse_error_r43 <- function(e, source_expression) {
   }
 
   if (column == 0L) {
-    line <- line - 1L
-    column <- nchar(source_expression$lines[[line]])
+    line_number <- line_number - 1L
   }
 
-  if (line < 1L || line > length(source_expression$lines)) {
+  if (line_number < 1L || line_number > length(source_expression$lines)) {
     # Safely handle invalid location info
-    line <- 1L
+    line_number <- 1L
     column <- 1L
   }
 
+  line <- fixup_line(source_expression$lines[[line_number]])
+  if (column == 0L || column > nchar(line) + 1L) {
```

Comment:
nit: it gives me some pause to see `nchar()` run 3 times, but I think it's fine b/c there's only one line here and this is only run once/file, is that right?

Reply:
This is only called once per file and only if a parse error occurred in that file.
Also, one of the `nchar()` calls could only be avoided by complicating the return value of `fixup_line()` to also return `nchar(line)`, or 0.

All in all, I don't really think this is worth the effort.

Reply:
Agreed, thanks for confirming.